rosidl_generator_py/rosidl_generator_py/generate_py_impl.py

```diff
@@ -73,6 +76,18 @@ def generate_py(generator_arguments_file, typesupport_impls):
         idl_file = parse_idl_file(locator)
         idl_content.elements += idl_file.content.elements
 
+    # NOTE(sam): remove when a language specific name mangling is implemented
+    for message in idl_content.get_elements_of_type(Message):
+        for member in message.structure.members:
+            msg_name = message.structure.namespaced_type.name
+            if (keyword.iskeyword(member.name)):
+                print(('Member name "{}" in the message "{}" is a '
+                       'reserved keyword in Python and is not supported '
+                       'at the moment. Please use a different name.')
+                      .format(member.name, msg_name), file=sys.stderr)
+
+    # TODO(sam): add reseved keyword warnings for services and actions
```

Comment:
It would be better to do this now, but if you want to do it across more than one pull request I guess that would be ok too.